### PR TITLE
pgrep: Add `TerminalType::Unknown` and implement `Display` for `TerminalType`

### LINKED ISSUE #136: pgrep: `--terminal` doesn't support `?` as value
`pgrep` from procps-ng supports `?` as a value for `--terminal` to match processes that do not have a controlling terminal. Our implementation doesn't support this value.
```
$ pgrep --terminal=? konsole
1265
1599
1640
1681
1741

$ cargo run pgrep --terminal=? konsole
// no output
```

### PR DESCRIPTION
**Pull request by Krysztal112233**
fix #136

### CODE REVIEW

**Comment by Krysztal112233** [comment]
Hmm... I'm not quite sure what kind of commands I should add to test, can you give some examples please?

**Comment by cakebaker** [comment]
You could add something like `pgrep --terminal=? kthreadd` and `pgrep --terminal=tty1 kthreadd`. The first case should succeed and the other fail with your fix.

**Comment by Krysztal112233** [comment]
> You could add something like `pgrep --terminal=? kthreadd` and `pgrep --terminal=tty1 kthreadd`. The first case should succeed and the other fail with your fix.

Nah I think this doesn't work on my system :(

```shell
$ pgrep --terminal=tty1 kthreadd
$ echo $?
1
$ pgrep --terminal=? kthreadd
$ echo $?
1
```

**Comment by cakebaker** [comment]
Hm, are there any PIDs returned with `pgrep --terminal=?`?

**Comment by Krysztal112233** [comment]
> Hm, are there any PIDs returned with `pgrep --terminal=?`?

After my investigation I realized that it was my procps version that was too low to support this syntax, and now I can get the output.

Now our implementation also works, but there are some problems: some high-privileged pid can't get to the terminal it belongs to, but the GNU implementation can.

```shell
❯ cargo run pgrep --terminal '?'
2977
2985
2986
2989
2992
2993
...
```

```shell
❯ pgrep --terminal=?
1
2
3
4
5
6
8
10
11
12
13
14
...
2977
2985
2986
2989
2992
2993
...
```

I don't know what to do about the above point for now, but for the `TerminalType` enumeration for `? ` support is now complete.

**Comment by Krysztal112233** [comment]
And, is `TerminalType::Orphan` perhaps better than `TerminalType::Unknow`?

**Comment by cakebaker** [comment]
I think `TerminalType::Unknown` is better.

File: src/uu/pgrep/src/process.rs [hunk 1]
```diff
@@ -302,14 +323,18 @@ impl ProcessInformation {
...
 
-        let result = Rc::new(
-            fs::read_dir(path)?
-                .flatten()
-                .filter(|it| it.path().is_symlink())
-                .flat_map(|it| fs::read_link(it.path()))
-                .flat_map(TerminalType::try_from)
-                .collect::<HashSet<_>>(),
-        );
+        let mut result = fs::read_dir(path)?
```
Comment (cakebaker) [inline src/uu/pgrep/src/process.rs:326]: I think you have to handle a permission error here and use a `TerminalType::Unknown` in such a case.

**Pull request merged**